psi4/src/psi4/dlpno/mp2.cc

```diff
@@ -0,0 +1,1412 @@
+            X_pno[ji] = X_pno[ij];
+            e_pno[ji] = e_pno[ij];
+            n_pno[ji] = n_pno[ij];
+            de_pno[ji] = de_pno_ij;
+        }
+    }
+
+    int pno_count_total = 0, pno_count_min = nbf, pno_count_max = 0;
+    de_pno_total_ = 0.0;
+    for (int ij = 0; ij < n_lmo_pairs; ++ij) {
+        pno_count_total += n_pno[ij];
+        pno_count_min = std::min(pno_count_min, n_pno[ij]);
+        pno_count_max = std::max(pno_count_max, n_pno[ij]);
+        de_pno_total_ += de_pno[ij];
+    }
+
+    outfile->Printf("  \n");
+    outfile->Printf("    Natural Orbitals per Local MO pair:\n");
+    outfile->Printf("      Avg: %3d NOs \n", pno_count_total / n_lmo_pairs);
+    outfile->Printf("      Min: %3d NOs \n", pno_count_min);
+    outfile->Printf("      Max: %3d NOs \n", pno_count_max);
+    outfile->Printf("  \n");
+    outfile->Printf("    PNO truncation energy = %.12f\n", de_pno_total_);
+
+#pragma omp parallel for schedule(static, 1)
+    for (int ij = 0; ij < n_lmo_pairs; ++ij) {
+        Tt_iajb[ij] = T_iajb[ij]->clone();
+        Tt_iajb[ij]->scale(2.0);
+        Tt_iajb[ij]->subtract(T_iajb[ij]->transpose());
+    }
+}
+
+void DLPNOMP2::pno_overlaps() {
+    int n_lmo_pairs = ij_to_i_j.size();
+    int naocc = nalpha_ - basisset_->n_frozen_core(options_.get_str("FREEZE_CORE"), molecule_);
+
+    S_pno_ij_kj.resize(n_lmo_pairs);
+    S_pno_ij_ik.resize(n_lmo_pairs);
+
+#pragma omp parallel for schedule(static, 1)
+    for (int ij = 0; ij < n_lmo_pairs; ++ij) {
+        int i, j;
+        std::tie(i, j) = ij_to_i_j[ij];
+        int ji = ij_to_ji[ij];
+
+        if (n_pno[ij] == 0) continue;
+        if (i < j) continue;
+
+        S_pno_ij_kj[ij] = std::vector<SharedMatrix>(naocc);
+        S_pno_ij_ik[ij] = std::vector<SharedMatrix>(naocc);
+
+        for (int k = 0; k < naocc; ++k) {
+            int kj = i_j_to_ij[k][j];
+            if (kj != -1 && i != k && fabs(F_lmo->get(i, k)) > options_.get_double("F_CUT") && n_pno[kj] > 0) {
+                S_pno_ij_kj[ij][k] = submatrix_rows_and_cols(S_pao, lmopair_to_paos[ij], lmopair_to_paos[kj]);
+                S_pno_ij_kj[ij][k] = linalg::triplet(X_pno[ij], S_pno_ij_kj[ij][k], X_pno[kj], true, false, false);
+            }
+
+            int ik = i_j_to_ij[i][k];
+            if (ik != -1 && j != k && fabs(F_lmo->get(k, j)) > options_.get_double("F_CUT") && n_pno[ik] > 0) {
+                S_pno_ij_ik[ij][k] = submatrix_rows_and_cols(S_pao, lmopair_to_paos[ij], lmopair_to_paos[ik]);
+                S_pno_ij_ik[ij][k] = linalg::triplet(X_pno[ij], S_pno_ij_ik[ij][k], X_pno[ik], true, false, false);
+            }
+        }
+
+        if (i > j) {
+            S_pno_ij_kj[ji] = S_pno_ij_ik[ij];
+            S_pno_ij_ik[ji] = S_pno_ij_kj[ij];
+        }
+    }
+}
+
+void DLPNOMP2::lmp2_iterations() {
+    int n_lmo_pairs = ij_to_i_j.size();
+    int naocc = nalpha_ - basisset_->n_frozen_core(options_.get_str("FREEZE_CORE"), molecule_);
```

Comment:
Why not get this information from `nfrzc()`? This comment applies everywhere you call `BasisSet.n_frozen_core`.